app/src/main/AndroidManifest.xml

```diff
@@ -50,6 +50,7 @@
             android:theme="@style/AppTheme.NoActionBar" />
         <activity
             android:name=".activities.BluetoothConnectionActivity"
+            android:launchMode = "singleInstance"
```

Comment:
In this case I think it is really necessary. I tried to find how it could come that the activity is started twice, and could not find out why. It doesn't tell in the logs and using finish just goes to the same "second" activity.
While googleing I found out that others had the same problem and resolved it by using this singleinstance.